src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -148,9 +148,430 @@
             value = "";
             break;
     }
-
     return strdup(value);
 }
+
+// invariant character definitions
+#define CHAR_CURRENCY ((char)0x00A4)   // international currency
+#define CHAR_SPACE ((char)0x0020)      // space
+#define CHAR_NBSPACE ((char)0x00A0)    // no-break space
+#define CHAR_DIGIT ((char)0x0023)      // '#'
+#define CHAR_MINUS ((char)0x002D)      // '-'
+#define CHAR_PERCENT ((char)0x0025)    // '%'
+#define CHAR_OPENPAREN ((char)0x0028)  // '('
+#define CHAR_CLOSEPAREN ((char)0x0029) // ')'
+#define CHAR_ZERO ((char)0x0030)       // '0'
+
+/*
+Function:
+NormalizeNumericPattern
+
+Returns a numeric string pattern in a format that we can match against the
+appropriate managed pattern.
+*/
+static char* NormalizeNumericPattern(const char* srcPattern, int isNegative)
+{
+    int iStart = 0;
+    int iEnd = strlen(srcPattern);
+
+    // ';'  separates positive and negative subpatterns. 
+    // When there is no explicit negative subpattern,
+    // an implicit negative subpattern is formed from the positive pattern with a prefixed.
+    char * ptrNegativePattern = strrchr(srcPattern,';');
+    if (ptrNegativePattern)
+    {
+        int32_t iNegativePatternStart = ptrNegativePattern - srcPattern;
+        if (isNegative)
+        {
+            iStart = iNegativePatternStart + 1;
+        }
+        else
+        {
+            iEnd = iNegativePatternStart - 1;
+        }
+    }
+
+    int index = 0;
+    int minusAdded = false;
+    int digitAdded = false;
+    int currencyAdded = false;
+    int spaceAdded = false;
```

Comment:
nit: maybe `index` `digitAdded` `currencyAdded` `spaceAdded` should all be moved down closer to where they're used?